Eatery/Controllers/Eateries/Campus/CampusEateryMealTableViewController.swift

```diff
@@ -63,20 +77,24 @@ class CampusEateryMealTableViewController: UITableViewController {
             // display menu items (of the only "dining station") as a table
             return item.value.count
         } else {
-            // display the menu items
-            return menu.data.count
+            // display the menu items with headers
+            let sortedMenu = Sort.sortMenu(menu.data.map { ($0, $1) })
+            return sortedMenu[section].1.count + 1
         }
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         guard let menu = menu else {
             return emptyMenuCell(in: tableView, forRowAt: indexPath)
         }
-
         if menu.data.count == 1, eatery.eateryType != .dining {
             return menuItemCell(in: tableView, forRowAt: indexPath)
         } else {
-            return diningStationsCell(in: tableView, forRowAt: indexPath)
+            if indexPath.item == 0 {
+                return diningStationsHeaderCell(in: tableView, forRowAt: indexPath)
+            } else {
+                return diningStationsCell(in: tableView, forRowAt: indexPath)
+            }
```

Comment:
I think instead of making the header a cell here, you can consider using `tableView(_:viewForHeaderInSection:)`